src/compiler/checker.ts

```diff
@@ -6877,7 +6876,40 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                     let typeArgumentNodes: readonly TypeNode[] | undefined;
                     if (typeArguments.length > 0) {
-                        const typeParameterCount = (type.target.typeParameters || emptyArray).length;
+                        let typeParameterCount: number;
+                        if (type.target.typeParameters) {
+                            typeParameterCount = Math.min(type.target.typeParameters.length, typeArguments.length);
+
+                            // Maybe we should do this for more types, but for now we only elide type arguments that are
+                            // identical to their associated type parameters' defaults for `Iterable`, `IterableIterator`,
+                            // `AsyncIterable`, and `AsyncIterableIterator` to provide backwards-compatible .d.ts emit due
+                            // to each now having three type parameters instead of only one.

```

Comment:
Realistically, this is all about what the type "looks" like, right? Or are we really needing to depend on this emit still emitting the "old thing" to continue to work right for downstream users of d.ts files?

Reply:
Consider an application that is currently built using TS 5.5 that depends on a library that is also being built using TS 5.5. The library chooses to upgrade its version of TypeScript to 5.6, and the new emit produces `Iterable<T, any, any>` where it previously produced `Iterable<T>`. If the library makes no other code changes and publishes their new build, the application cannot take the update without *also* updating TypeScript because the expected arity of `Iterable` has changed.
The library can't just ship a forward declaration of `interface Iterable<T, TReturn = any, TNext = any> {}`, because our checker performs arity checks on built-ins, and the application might not be running with `skipLibCheck`, so one way or another they would have errors they cannot silence without possibly giving up the dependency verification that `skipLibChecks: false` offers.